common/components/register_model/spec.yaml

```diff
@@ -0,0 +1,63 @@
+$schema: https://azuremlschemas.azureedge.net/latest/commandComponent.schema.json
+name: register_model
+version: 0.0.1
+type: command
+
+is_deterministic: True
+
+display_name: Register model
+description:
+  Register a model to a workspace or a registry. The component works on compute with [MSI](https://learn.microsoft.com/en-us/azure/machine-learning/how-to-create-manage-compute-instance?tabs=python) attached.
+
+environment: azureml:python-sdk-v2:1
+
+code: ../../src/model_registration
+command: >-
+  python register.py
+  --model_path ${{inputs.model_path}}
+  $[[--model_type ${{inputs.model_type}}]]
+  $[[--model_name ${{inputs.model_name}}]]
+  $[[--registry_name ${{inputs.registry_name}}]]
+  $[[--model_info_path ${{inputs.model_info_path}}]]
+  $[[--model_description "${{inputs.model_description}}"]]
+  --registration_details ${{outputs.registration_details}}
+
+inputs:
+  model_name:
+    type: string
+    optional: true
+    description: Model name to use in the registration. If name already exists, the version will be auto incremented
+
+  model_type:
+    type: string
+    optional: true
+    enum:
+      - custom_model
+      - mlflow_model
+    default: mlflow_model
+    description: Model type
+
+  model_description:
+    type: string
+    optional: true 
+    description: Description of the model that will be shown in AzureML registry or workspace
+
+  registry_name:
+    type: string
+    optional: true
+    description: Name of the AzureML asset registry where the model will be registered. Model will be registered in a workspace if this is unspecified
+
+  model_path:
+    type: uri_folder
+    optional: false
+    description: Path to the model directory
+
+  model_info_path:
+    type: uri_file
+    optional: true
+    description: A JSON file to provide model registration specific settings (name, type, description, tags, properties). This will take priority over values provided through component inputs
+
+outputs:
+  registration_details:
+    type: uri_file
+    description: Text File into which model registration details will be written
```

Comment:
> Text File into which model registration details will be written [](http://example.com/codeflow?start=16&length=64)

Is it possible to make the output the "registered model" itself?

Where we refer to the model through the registry URI?

Reply:
Created a task for this: https://github.com/Azure/azureml-assets/issues/419 for tracking